extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -1016,22 +1091,66 @@ static class StringTransfer implements TransferObject<Binary[]> {
 
         private final ChunkSource.GetContext context;
         private ObjectChunk<String, Values> chunk;
-        private final Binary[] buffer;
+        private Binary[] buffer;
         private final ColumnSource<?> columnSource;
 
+        /**
+         * Stores the maximum number of values in a single page
+         */
+        private final int maxValuesPerPage;
+
+        /**
+         * Stores the maximum size of data in a single page. For strings, we don't know in advance what the length of
+         * each entry would be, therefore we need to keep track of both maximum size and number of values.
+         */
+        private final int targetPageSize;
+
+        /**
+         * Total number of chunks buffered so far
+         */
+        private int numChunksBuffered;
 
-        StringTransfer(ColumnSource<?> columnSource, int targetSize) {
+
+        StringTransfer(ColumnSource<?> columnSource, int maxValuesPerPage, int targetPageSize) {
             this.columnSource = columnSource;
-            this.buffer = new Binary[targetSize];
-            context = this.columnSource.makeGetContext(targetSize);
+            this.buffer = new Binary[maxValuesPerPage];
+            context = this.columnSource.makeGetContext(maxValuesPerPage);
+            numChunksBuffered = 0;
+            this.maxValuesPerPage = maxValuesPerPage;
+            this.targetPageSize = targetPageSize;
         }
 
         @Override
-        public void propagateChunkData() {
-            for (int i = 0; i < chunk.size(); i++) {
-                String value = chunk.get(i);
-                buffer[i] = value == null ? null : Binary.fromString(value);
+        public boolean hasMoreDataToBuffer() {
+            return ((chunk != null) && (numChunksBuffered < chunk.size()));
+        }
+
+        public int bufferFetchedData() {
+            if (numChunksBuffered != 0) {
+                // Clear any old data in the buffer from previous iterations
+                buffer = new Binary[maxValuesPerPage];
             }
+            int bufferedDataSize = 0;
+            int bufferIdx = 0; // Stores the number of chunks buffered in this iteration
+            while (numChunksBuffered < chunk.size()) {
+                final String value = chunk.get(numChunksBuffered);
+                if (value == null) {
+                    buffer[bufferIdx++] = null;
+                } else {
+                    Binary binaryVal = Binary.fromString(value);
+                    if (binaryVal.length() > targetPageSize) {
+                        throw new UncheckedDeephavenException(String.format("Cannot fit string data of size %d in a " +
+                                "single page of maximum size %d", value.length(), targetPageSize));
+                    }
+                    if (bufferedDataSize + binaryVal.length() > targetPageSize) {
+                        break;
```

Comment:
As you noted in our call, this termination  may cause truncation when a vector of strings is to be written  that exceeds the target page size.  We need to handle vectors/arrays of strings differently and be accept potential OOM and oversize pages.